js/piwik.js

```diff
@@ -7268,29 +7281,31 @@ if (typeof window.Piwik !== 'object') {
              * @param float price Item's display price, not use in standard Piwik reports, but output in API product reports.
              */
             this.setEcommerceView = function (sku, name, category, price) {
-                if (!isDefined(category) || !category.length) {
+                if (isNumberOrHasLength(category)) {
+                    category = String(category);
+                }
+                if (!isDefined(category) || category === null || category === false || !category.length) {
                     category = "";
                 } else if (category instanceof Array) {
                     category = JSON_PIWIK.stringify(category);
                 }
 
                 customVariablesPage[5] = ['_pkc', category];
 
-                if (isDefined(price) && String(price).length) {
+                if (isDefined(price) && price !== null && price !== false && String(price).length) {
```

Comment:
BTW I'm not sure it's on purpose we allow tracking price when there is only category set? Could create another issue for it... maybe it's on purpose ... @mattab

Reply:
BTW `String(false).length` would return `5` which is why I added specific check for false and null etc.

Reply:
Can't remember if it's on purpose or not, but seems one should be able to track price even without a category set @tsteur

Reply:
Actually, if I see this right you can track just a category, or a category with a price. Not sure why you would track a category with a price but not an SKU or name... sounds to me like price makes only sense when also name or SKU is set?